src/test.ts

```diff
@@ -38,21 +250,147 @@ export function testStreamFromObject<A>(object: {
   return new TestStream(semanticStream);
 }
 
-class TestFuture<A> extends Future<A> {
-  constructor(private semanticFuture: SemanticFuture<A>) {
+export function assertStreamEqual<A>(s1: Stream<A>, s2: Stream<A>): void;
+export function assertStreamEqual<A>(
+  s1: Stream<A>,
+  s2: {
+    [time: number]: A;
+  }
+): void;
+export function assertStreamEqual<A>(s1: Stream<A>, s2: A[]): void;
+export function assertStreamEqual<A>(s1: Stream<A>, s2): void {
+  const s2_ = isStream(s2)
+    ? s2
+    : Array.isArray(s2)
+    ? testStreamFromArray(s2)
+    : testStreamFromObject(s2);
+  assert.deepEqual(s1.model(), s2_.model());
+}
+
+// Behavior
+
+export type BehaviorModel<A> = (time: Time) => A;
+
+declare module "./behavior" {
+  interface Behavior<A> {
+    model(): BehaviorModel<A>;
+  }
+}
+
+MapBehavior.prototype.model = function() {
+  const g = this.parent.model();
+  return (t) => this.f(g(t));
+};
+
+ConstantBehavior.prototype.model = function() {
+  return (_) => this.last;
+};
+
+FunctionBehavior.prototype.model = function() {
+  return (t: number) => this.f(t);
+};
+
+time.model = () => (t: Time) => t;
+
+ScanBehavior.prototype.model = function<A, B>(this: ScanBehavior<A, B>) {
+  const stream = this.source.model();
+  return (t1) =>
+    testBehavior<B>((t2) =>
+      stream
+        .filter(({ time }) => t1 <= time && time <= t2)
+        .map((o) => o.value)
+        .reduce((acc, cur) => this.f(cur, acc), this.initial)
+    );
+};
+
+class TestBehavior<A> extends Behavior<A> {
+  constructor(private semanticBehavior: BehaviorModel<A>) {
     super();
   }
-  pushS(t: number, val: A): void {
-    throw new Error("You cannot push to a TestFuture");
+  /* istanbul ignore next */
+  update(_t: number): A {
+    throw new Error("Test behavior never updates");
   }
-  semantic(): SemanticFuture<A> {
-    return this.semanticFuture;
+  model(): BehaviorModel<A> {
+    return this.semanticBehavior;
   }
-  push(a: A): void {
-    throw new Error("You cannot push to a TestFuture");
+}
+
+export function testBehavior<A>(b: (time: number) => A): Behavior<A> {
+  return new TestBehavior(b);
+}
+
+/**
+ * Takes a behavior created from test data, a point in timer and returns the
+ * behaviors value at that point in time.
+ */
+export function testAt<A>(t: number, b: Behavior<A>): A {
+  return b.model()(t);
+}
+
+// * Now
+
+type NowModel<A> = { value: A; mocks: any[] };
+
+declare module "./now" {
+  interface Now<A> {
+    test(mocks: any[], t: Time): NowModel<A>;
   }
 }
 
-export function testFuture<A>(time: number, value: A): Future<A> {
-  return new TestFuture({ time, value });
+OfNow.prototype.test = function test<A>(mocks: any[], _t: Time): NowModel<A> {
```

Comment:
Here you actually name the function, the others are anonymous.

Reply:
Good catch. I've made them all anonymous.